### PR TITLE
AWS SAM -- Readme Documentation + Example

### PR DESCRIPTION
**Pull request by Dgarc359**
*Issue #, if available:*
#149
*Description of changes:*
Update readme with reference to new example sam project in example directory

Apologies that #152 got closed by accident. Mistake on my part.

My formatter also removed trailing spaces from the Readme. I can push another commit to clean that up if it's necessary.

Tagging @richarddavison for visibility

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by richarddavison — CHANGES_REQUESTED** [review]
Thanks for the PR. Great start! Can we make this a bit less opinionated? For example, remove all the linting, jest/tests, eslint etc and focus on a minimum working example.
Even though it's pretty cool to always pull the latest version via the Makefile it would be a bit simpler to just point to an existing zip file as a layer. That way, we can also avoid building with docker, and don't require users to have Make installed.

**Comment by Dgarc359** [comment]
> Thanks for the PR. Great start! Can we make this a bit less opinionated? For example, remove all the linting, jest/tests, eslint etc and focus on a minimum working example. Even though it's pretty cool to always pull the latest version via the Makefile it would be a bit simpler to just point to an existing zip file as a layer. That way, we can also avoid building with docker, and don't require users to have Make installed.

Latest change applies the above. Worth noting that the example no longer pulls the llrt zip during build and expects the user to place the zip file in the root of the `llrt` directory. I saw a similar pattern being applied to the example in the `infrastructure` directory. Thank you for the feedback :)

File: example/llrt-sam/.gitignore [hunk 1]
```diff
@@ -0,0 +1,634 @@
+
```
Comment (richarddd) [inline example/llrt-sam/.gitignore:1]: Can we also make this bare minimum?

File: example/llrt-sam/template.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Globals:
+  Function:
+    Timeout: 3
+
+Resources:
+
+  LlrtLayer:
+    Type: AWS::Serverless::LayerVersion
+    Properties:
+      ContentUri: ../../llrt-lambda-x64.zip
```
Comment (richarddd) [inline example/llrt-sam/template.yaml:18]: ARM is actually faster, so we should default to that

Comment (richarddd) [reply example/llrt-sam/template.yaml:18]: Also remember set `CompatibleArchitectures` to `arm64`

File: example/llrt-sam/template.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      CompatibleRuntimes:
+        - provided.al2023
+  HelloWorldFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: hello-world/
+      Handler: app.lambdaHandler
+      Runtime: provided.al2023
+      Architectures:
+        - x86_64
```
Comment (richarddd) [inline example/llrt-sam/template.yaml:28]: arm64

**Review by richarddd — CHANGES_REQUESTED** [review]
Almost there! Just a couple of minor changes and it LGTM

File: example/llrt-sam/template.yaml (hunk 2 above)
Comment (kaihendry) [reply example/llrt-sam/template.yaml:18]: Wasn't any faster when I tried arm64 https://github.com/kaihendry/llrt-aws-sam-example/actions/runs/7907126599/job/21583462311#step:7:20 vs x64 https://github.com/kaihendry/llrt-aws-sam-example/actions/runs/7907082851/job/21583326692#step:7:20 thought it probably wasn't the best test

Comment (richarddavison) [reply example/llrt-sam/template.yaml:18]: That probably due to caches, Lambda uses a tiered cache that gets a bit better after a while. Also ARM is lower cost.  It's not a major difference in performance but you should experience slightly faster cold starts at least (at lower cost) :)

**Review by richarddavison — APPROVED** [review]

File: example/llrt-sam/template.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        - arm64
+      Layers:
+        - !Ref LlrtLayer
+    Metadata: # Manage esbuild properties
+      BuildMethod: esbuild
+      BuildProperties:
+        External:
+          - '@aws-sdk'
+          - '@smithy'
+          - uuid
```
Comment (kaihendry) [inline example/llrt-sam/template.yaml:39]: Does this mean these are already present in the layer / runtime?

Comment (richarddavison) [reply example/llrt-sam/template.yaml:39]: Yes! See https://github.com/awslabs/llrt?tab=readme-ov-file#using-aws-sdk-v3-with-llrt

**Pull request merged**